docs/design/persistent-storage.md

```diff
@@ -0,0 +1,235 @@
+# Persistent Stores
+
+This document proposes a model for managing persistent, cluster-scoped storage for applications requiring long lived data.
+
+### tl;dr
+
+`PersistentStore` objects contain a VolumeSource with a lifecycle greater than the pods it is used by.  Many types of VolumeSources can be implemented as volume plugins, with each implementing an interface for use within a controller.  A pod with persistence requirements (specifically pod.spec.volumes.volume.persistenceRequirements) is bound to a matching VolumeSource from a pool of available stores.  PersistentStore objects give that VolumeSource longevity.
```

Comment:
Yes, it is a volume.  I keep using VolumeSource because that's the field, but they are volumes.  

A single PersistentStore instance is one volume (i.e, an actual gce_pd) in the cluster.  

The store controllers also use a volume (not a store) as a template.  If the volume plugin supports dynamic provisioniong, the controller fills the pool with that type of volume as instances of PersistentStore.  The pool is still managed by the controller, so providers that do not support dynamic provisioning must be added via the API.

@bgrant0607 was referencing prototyping in another issue.  An infrastructure's distinct set of volumes would be the prototypes.